base/linalg/generic.jl

```diff
@@ -417,7 +417,49 @@ istriu(x::Number) = true
 istril(x::Number) = true
 isdiag(x::Number) = true
 
-linreg{T<:Number}(X::AbstractVector{T}, y::AbstractVector{T}) = [ones(T, size(X,1)) X] \ y
+#TODO: lookup where all the documentation for linreg is coming from, including the example
+# mentioning PyPlot etc
+"""
+    linreg(x, y)
+
+Find the Ordinary Least Squares best linear fit between the numerical vectors `x` and `y`
+
+Inputs:
+
+- `x::AbstractVector{T <: Number}` - vector of independent variables
+- `y::AbstractVector{S <: Number}` - vector of dependent variables
+
+Output:
+
+- `[b0, b1]` - the coefficients for the line of best fit such that yhat = b0 + b1*x
+
+See also:
+
+`\\`, `cov`, `std`, `mean`
+
+"""
+function linreg{T <: Number, S <: Number}(x::AbstractVector{T}, y::AbstractVector{S})
+    # Least squares given
+    # Y = b0 + b1*X
+    # where
+    # b1 = cov(X, Y)/var(X)
+    # b0 = mean(Y) - b1*mean(X)
+    # Calculate the cov and var using the fewest iterations of the inputs possible
+    n = length(x)
+    @assert n == length(y)
+    mx = mean(x) # need to know this at each step do the implicit loop over x first
+    tot_dx = 0.0
+    tot_y = 0.0
+    tot_xy = 0.0
+    for i in 1:n
```

Comment:
Do I just put the `@inbounds` in front of the for loop? Doesn't seem to change any of my timings, but seems harmless.